src/Core/SharedKernels/DataCollection/DataCollection/Aggregates/IStatefulInterview.cs

```diff
@@ -213,5 +213,13 @@ public interface IStatefulInterview
 
         InterviewSimpleStatus GetInterviewSimpleStatus(bool includingSupervisorEntities);
         IEnumerable<Identity> GetNeighboringQuestionIdentities(Identity identity);
+
+        IEnumerable<Guid> RunAndGetFailCriticalityConditions();
+        
+        IEnumerable<Identity> GetAllUnansweredCriticalQuestions();
+        IEnumerable<Identity> GetAllUnansweredQuestions();
+        
+        string GetCriticalityConditionMessage(Guid criticalityConditionId);
+        bool HasCriticalFeature();

```

Comment:
IsCriticalitySupported?
Do we need it here? It looks like it's a questionnaire responsibility to handle it.